experimental/Schemes/singular_locus.jl

```diff
@@ -3,28 +3,115 @@ export singular_locus, singular_locus_reduced
 export reduced_scheme
 export is_smooth
 
-MPAnyQuoRing = Union{MPolyQuoLocalizedRing, 
-                MPolyQuo
-               }
 
-MPAnyNonQuoRing = Union{MPolyRing, MPolyLocalizedRing
-                  }
+#######################################################################
+# Functionality for smoothness/singular locus
+#######################################################################
 
-### TODO: The following two functions need to be made type-sensitive 
+#######################################################################
+# MOVE TO Methods
+#######################################################################
+
+### TODO: The following two functions (singular_locus, 
+###       singular_locus_reduced) need to be made type-sensitive 
 ###       and reduced=true needs to be set automatically for varieties 
 ###       as soon as not only schemes, but also varieties as special 
 ###       schemes have been introduced in OSCAR
-function singular_locus(X::AbsSpec{<:Ring, <:MPAnyQuoRing})
+### TODO: Make singular locus also available for projective schemes and
+###       for covered schemes (the workhorse is already there...).
+ 
+@doc Markdown.doc"""
+    singular_locus(X::AbsSpec{<:Field, <:MPolyAnyRing})
+
+Returns the singular locus of $X$ as a subscheme of $X$, where $X$ is a
+scheme over a field $k$. For computing the singular locus of the 
+reduced scheme induced by $X$, please use 'singular_locus_reduced'.
+
+Currently this command is available for affine schemes and space germs.
+TODO: Covered schemes, projective schemes
```

Comment:
Please follow
https://docs.julialang.org/en/v1/manual/documentation/
when writing documentation.
Note also the backticks " ` " vs " '  "
Here I would change it as follows.
```suggestion
@doc Markdown.doc"""
    singular_locus(X::AbsSpec{<:Field, <:MPolyAnyRing}) -> AbsSpec

Return the singular locus of `X`.

For computing the singular locus of the reduced scheme induced by $X$, please use `singular_locus_reduced`.

Currently this command is available for affine schemes and space germs.
TODO: Covered schemes, projective schemes
```